CollapseLauncher/Classes/GameManagement/GamePlaytime/RegistryClass/CollapsePlaytime.cs

```diff
@@ -226,5 +248,151 @@ public void AddMinute()
+        
+        // value store
+        private string? _jsonDataDb;
+        private double? _totalTimeDb;
+        private double? _lastPlayedDb;
+        private int?    _unixStampDb;
+        
+        // Key names
+        private string KeyPlaytimeJson => $"{_gameVersion.GameType.ToString()}-{_gameVersion.GameRegion}-pt-js";
+        private string KeyTotalTime    => $"{_gameVersion.GameType.ToString()}-{_gameVersion.GameRegion}-pt-total";
+        private string KeyLastPlayed   => $"{_gameVersion.GameType.ToString()}-{_gameVersion.GameRegion}-pt-lastPlayed";
+        private string KeyLastUpdated  => $"{_gameVersion.GameType.ToString()}-{_gameVersion.GameRegion}-pt-lu";
+        
+
+        #region Sync Methods
+        /// <summary>
+        /// Sync from/to DB at init
+        /// </summary>
+        /// <returns>true if require refresh, false if dont.</returns>
+        public async ValueTask<(bool IsUpdated, CollapsePlaytime? PlaytimeData)> DbSync()
+        {
+            LogWriteLine("[CollapsePlaytime::DbSync] Starting sync operation...", LogType.Default, true);
+            try
+            {
+                // Pull values from DB
+                await UpdatePlaytime_Database_Pull();
+                if (!_isDbPullSuccess) 
+                {
+                    LogWriteLine("[CollapsePlaytime::DbSync] Database pull failed, skipping sync~", LogType.Error);
+                    return (false, null); // Return if pull failed
+                }
+
+                // Compare unix stamp from config
+                var unixStampLocal = Convert.ToInt32(DbConfig.GetConfig(KeyLastUpdated).ToString());
+                if (_unixStampDb == unixStampLocal) 
+                {
+                    LogWriteLine("[CollapsePlaytime::DbSync] Sync stamp equal, nothing needs to be done~", LogType.Default, true);
+                    return (false, null); // Do nothing if stamp is equal
+                }
+
+                // When Db stamp is newer, sync from Db
+                if (_unixStampDb > unixStampLocal)
+                {
+                    if (string.IsNullOrEmpty(_jsonDataDb))
+                    {
+                        LogWriteLine("[CollapsePlaytime::DbSync] _jsonDataDb is empty, skipping sync~", default, true);
+                        return (false, null);
+                    }
+                    LogWriteLine("[CollapsePlaytime::DbSync] Database data is newer! Pulling data~", LogType.Default, true);
+                    CollapsePlaytime? playtimeInner = _jsonDataDb.Deserialize(UniversalPlaytimeJSONContext.Default.CollapsePlaytime,
+                                                        new CollapsePlaytime());
+
+                    playtimeInner!.TotalPlaytime = TimeSpan.FromSeconds(_totalTimeDb ?? 0);
+                    if (_lastPlayedDb != null) playtimeInner.LastPlayed = BaseDate.AddSeconds((int)_lastPlayedDb);
+                    playtimeInner._registryRoot = _registryRoot;
```

Comment:
Needs also to pass _gameSettings to the new instance.

Reply:
fixed in https://github.com/CollapseLauncher/Collapse/pull/586/commits/c8456999ca89f52a457ff56f5c587e849e33e8c6